### PR TITLE
React capstone project - Stock Data webapp

### PR DESCRIPTION
**Pull request by LIBERCOSOFT**
## A Stock data Webapp for over 5000 stocks
- Consume Stock data API.
- Display list of stocks gotten from API.
- Display the details of each stock.
- Add test to components.
- Link all pages with react-router.
- Use Redux as state management.
- Use JEST for testing.

**Live Link** - https://lighthearted-druid-5c17e6.netlify.app/
**Video Link** - https://www.loom.com/share/46bd3f80bc7d4e488b20f8e7d0c485e7

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @LIBERCOSOFT, Kindly use the updated version  of the microverse [readme](https://github.com/microverseinc/readme-template) template for your readme, this is a requirement to get a review

**Comment by prolajumokeoni** [comment]
Hi @LIBERCOSOFT  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have the updated microverse  [readme](https://github.com/microverseinc/readme-template)
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/pages/Homepage.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            See All Stocks
+            {' '}
+            <i className="fa-sharp fa-solid fa-arrow-right" />
+            {' '}
+          </Link>
+        </div>
+      </div>
+      <h4>Filter By Sector:</h4>
+      <div className="filter__container">
+        {filterOptions.map((val) => (
```
Comment (FrederickMih) [inline src/pages/Homepage.js:82]: - [x] Based on the requirements of this project, which say that the Homepage should have a list of categories(Let's kindly display a list of stocks on the homepage :+1: ). These stocks should be filtered (We shall learn to JavaScript ES6 filter() method) using a given parameter(You can use the stock name). We shall use the **Filter** stateless component to filter data retrieved from the API.


NB: You can use a search field on the homepage. Inside the search field, a user can introduce the beginning of the name of a stock and the filter will get all stocks with those letters.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @LIBERCOSOFT, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

Screenshot from 2022-10-20 09-38-51

## Highlights :green_circle: 

- The app contains the Home and Detail page as requested :+1: 
- Redux has been used for state management :+1: 

## Suggested changes 

- [x] Kindly add unit tests for some pure functions in the Redux code.
- [x] App should have integration tests using React Testing Library(We can perform a mocking data fetching and Redux store connection)

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by ShadyShawkat — APPROVED** [review]
## Hi @LIBERCOSOFT, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 
- Your code is clean and well documented ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**